### PR TITLE
Added support for creating edge regions

### PR DESCRIPTION
**Pull request by juanpardo**
## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: croud/__main__.py [hunk 1]
```diff
@@ -543,6 +543,38 @@
                 "help": "List all available regions",
                 "resolver": regions_list,
             },
+            "create": {
+                "help": "Create a new region.",
+                "resolver": regions_create,
+                "extra_args": [
+                    Argument(
+                        "--aws-bucket", type=str, required=False,
```
Comment (turbo-ele) [inline croud/__main__.py:551]: but required in the api

**Review by turbo-ele — COMMENTED** [review]
Tests still missing, which you are I guess already aware of :)

File: croud/__main__.py [hunk 2]
```diff
@@ -543,6 +543,38 @@
...
+                        help="The AWS S3 bucket where cluster backups will be stored.",
+                    ),
+                    Argument(
+                        "--aws-region", type=str, required=False,
+                        help="The AWS region where the S3 bucket for cluster"
```
Comment (SStorm) [inline croud/__main__.py:556]: Why does the bucket region matter?

Comment (juanpardo) [reply croud/__main__.py:556]: I don't know why, but the Brain table holds that information and it's expecting it.

File: croud/__main__.py (hunk 1 above)
Comment (juanpardo) [reply croud/__main__.py]: Just aligned these to the api expectations

**Review by turbo-ele — COMMENTED** [review]
looks good, but what about the tests?

Also, please update the changelog and documentation.

**Comment by SStorm** [comment]
This LGTM, but the build is failing and there's no conflicts :/

**Review by lukasbals — COMMENTED** [review]
Looks good to me. But of course, I wouldn't merge it until the backend endpoint is released, not to block the croud to be released in the meantime.

**Comment by juanpardo** [comment]
Solved the conflicts, rebased and squashed :)

File: croud/regions.py [hunk 3]
```diff
@@ -37,3 +37,31 @@ def regions_list(args: Namespace) -> None:
...
+        "aws_region": args.aws_region,
+        "description": args.description,
+        "provider": args.provider,
+    }
+
+    # Add optional parameters only when present
+    if args.name:
+        body.setdefault("name", args.name)
+    if args.org_id:
+        body.setdefault("organization_id", args.org_id)
```
Comment (turbo-ele) [inline croud/regions.py:58]: imo, there's no need to use setdefault?

```suggestion
        body["name"] = args.name
    if args.org_id:
        body["organization_id] = args.org_id
```

File: docs/commands/regions.rst [hunk 4]
```diff
@@ -33,3 +33,31 @@ Example
...
+.. note::
+
+   Only superusers can create regions.
+
+Example
+=======
+
+.. code-block:: console
+
+   sh$ croud regions create --org-id 926d4e6b-5967-4ab4-8f46-82a90150ab31 --description "Edge region" --provider EDGE --aws-bucket backup-bucket --aws-region eu-west-1 --sudo
```
Comment (turbo-ele) [inline docs/commands/regions.rst:58]: is this an artificial org-id?

**Review by turbo-ele — APPROVED** [review]
lgtm

**Pull request merged**